scipy/stats/_sensitivity_analysis.py

```diff
@@ -591,7 +588,13 @@ def sobol_indices(
                 "the number of output."
             )
 
-        f_B, f_AB = np.atleast_2d(func(B), func(AB))
+        def funcAB(AB):
+            m, m, n = AB.shape
+            AB = np.moveaxis(AB, 0, -1).reshape(m, n*m)
+            f_AB = func(AB)
+            return np.moveaxis(f_AB.reshape(-1, n, m), -1, 0)
```

Comment:
There might be a more concise way to do these, but this is how I think about these operations. The way its done ensures that we don't run into issues like https://github.com/scipy/scipy/pull/17628#discussion_r1063713851